### DISCUSSION TITLE
help wanted: stream-sorting-excludes in the config-file

### DISCUSSION THREAD

**Discussion by LinuxOpa** [streamlink/streamlink#3831] [Q&A] [answered]
hi all.

how to use --stream-sorting-excludes in the config file?
i tried something but nothing worked.
i tried stream-sorting-excludes >720p in the config-file
to stream nothing greater 720p but it doesnt work,
it only shows "Available streams: 360p_http, 720p_http, 1080p_http, 144p (worst), 270p, 360p, 540p, 544p, 720p, 1080p"
how is the correct syntax?

thanks in advance

**Comment by bastimeyer** [comment, marked as answer]
> i tried stream-sorting-excludes >720p in the config-file

Take a look at the CLI docs and the config file syntax:
https://streamlink.github.io/cli.html#syntax

> The config file is a simple text file and should contain one command-line option (omitting the dashes) per line in the format:
> > option=value

The syntax of the `--stream-sorting-excludes` argument is this:
https://streamlink.github.io/cli.html#cmdoption-stream-sorting-excludes

Which means this is how your config file should look like:
```
stream-sorting-excludes=>720p
```

`--loglevel debug` via the command line lets you know all the arguments and values which are set.

Remember that `--stream-sorting-excludes` will only change the `best` and `worst` stream name synonyms and will keep all the available streams. There are also the `best-unfiltered` and `worst-unfiltered` "hidden" stream name synonyms which won't be affected by the exclusion.

----

Examples:

Without `stream-sorting-excludes`:

```
$ cat "${XDG_CONFIG_HOME:-${HOME}/.config}/streamlink/config"

$ streamlink -l debug twitch.tv/esl_csgo
[cli][debug] OS:         Linux-5.12.13-2-git-x86_64-with-glibc2.33
[cli][debug] Python:     3.9.5
[cli][debug] Streamlink: 2.2.0+19.ge2ca35b
[cli][debug] Requests(2.25.1), Socks(1.7.1), Websocket(0.58.0)
[cli][debug] Arguments:
[cli][debug]  url=twitch.tv/esl_csgo
[cli][debug]  --loglevel=debug
[cli][info] Found matching plugin twitch for URL twitch.tv/esl_csgo
[plugins.twitch][debug] Getting live HLS streams for esl_csgo
[utils.l10n][debug] Language code: en_US
Available streams: audio_only, 160p (worst), 360p, 480p, 720p, 720p60, 1080p60 (best)
```

`stream-sorting-excludes=>720p`:

```
$ cat "${XDG_CONFIG_HOME:-${HOME}/.config}/streamlink/config"
stream-sorting-excludes=>720p

$ streamlink -l debug twitch.tv/esl_csgo
[cli][debug] OS:         Linux-5.12.13-2-git-x86_64-with-glibc2.33
[cli][debug] Python:     3.9.5
[cli][debug] Streamlink: 2.2.0+19.ge2ca35b
[cli][debug] Requests(2.25.1), Socks(1.7.1), Websocket(0.58.0)
[cli][debug] Arguments:
[cli][debug]  url=twitch.tv/esl_csgo
[cli][debug]  --loglevel=debug
[cli][debug]  --stream-sorting-excludes=['>720p']
[cli][info] Found matching plugin twitch for URL twitch.tv/esl_csgo
[plugins.twitch][debug] Getting live HLS streams for esl_csgo
[utils.l10n][debug] Language code: en_US
Available streams: audio_only, 160p (worst), 360p, 480p, 720p (best), 720p60, 1080p60
```

**Reply by LinuxOpa** [reply]
excact this is what i wanted!

THANK YOU!
